src/AWS.Logger.Core/Core/AWSLoggerCore.cs

```diff
@@ -662,23 +662,75 @@ private void LogLibraryServiceError(Exception ex, string serviceUrl = null)
         /// <summary>
         /// Write Exception details to the file specified with the filename
         /// </summary>
-        public static void LogLibraryError(Exception ex, string LibraryLogFileName)
+        public static void LogLibraryError(Exception? ex, string LibraryLogFileName)
         {
             try
             {
-                using (StreamWriter w = File.AppendText(LibraryLogFileName))
-                {
-                    w.WriteLine("Log Entry : ");
-                    w.WriteLine("{0}", DateTime.Now.ToString());
-                    w.WriteLine("  :");
-                    w.WriteLine("  :{0}", ex.ToString());
-                    w.WriteLine("-------------------------------");
-                }
+                StringBuilder sb = new StringBuilder(512);
+
+                sb.AppendLine("Log Entry : ");
+
+                sb.AppendLine(DateTime.Now.ToString());
+
+                sb.AppendLine("  :");
+
+                sb.Append("  :");
+                sb.AppendLine(ex.ToString());
+
+                sb.AppendLine("-------------------------------");
+
+                WriteToFile(LibraryLogFileName, sb.ToString());
             }
             catch (Exception e)
             {
                 Console.WriteLine("Exception caught when writing error log to file" + e.ToString());
             }
         }
+
+        static ReaderWriterLock locker = new ReaderWriterLock();
```

Comment:
Why use a ReaderWriter lock since you are never using it as a reader lock? You can switch to using `Monitor.TryEnter` and `Monitor.Exit`.